### PR TITLE
Fix 9053: Remove gSceneryGroundFlags by returning it within GameActionResult 

### PR DESCRIPTION
**Pull request by tharkne**
This should have removed and replaced all the references to gSceneryGroundFlags in the code as per #9053. Are there any other changes that should be made?

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
That looks good. You just need to do a clang format for the ci to be happy. Either follow the changes requested by the ci or do it programmatically.

Most IDEs can have clang format integrated in. Or as a plugin. You might find that beneficial. Then it's just a one button click to fix everything.

**Comment by tharkne** [comment]
OK, it seems everything is alright now.

**Comment by duncanspumpkin** [comment]
argh I mucked this up I need to delete my merge commit.

**Comment by duncanspumpkin** [comment]
Argh why do i have write permission im just making this worse.

**Comment by duncanspumpkin** [comment]
Sorry for butchering your PR with some silly commits. Its all ready to go now.

**Comment by duncanspumpkin** [comment]
Okay that should fix it. All looks to be working great. Will be ready to merge once ci's are green,

**Comment by tharkne** [comment]
Haha, thanks for your help. Good to go on the merge then?

File: src/openrct2/actions/SmallSceneryPlaceAction.hpp [hunk 1]
```diff
@@ -27,6 +27,29 @@
...
+    }
+    SmallSceneryPlaceActionResult(GA_ERROR error, rct_string_id message, uint8_t* args)
+        : GameActionResult(error, STR_CANT_POSITION_THIS_HERE, message, args)
+    {
+    }
+
+    uint8_t GroundFlags{ 0 };
+};
+
 DEFINE_GAME_ACTION(SmallSceneryPlaceAction, GAME_COMMAND_PLACE_SCENERY, GameActionResult)
```
Comment (duncanspumpkin) [inline src/openrct2/actions/SmallSceneryPlaceAction.hpp]: `SmallSceneryPlaceActionResult`

**Comment by duncanspumpkin** [comment]
Just noticed one minor mistake. The definition for small scenery place action needs to be changed to add in the new class. It won't effect how it works but it means some helpful functions work correctly. (Those helpful functions are not used right now. Hence why it still works.)

**Comment by tharkne** [comment]
Good catch! Must've left that one behind somehow.

**Pull request merged**